nmdc_automation/re_iding/tests/test_db_utils.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from nmdc_automation.re_iding.db_utils import (
+    get_collection_name_from_workflow_id,
+    fix_malformed_workflow_id_version,
+    fix_malformed_data_object_name,
+    fix_malformed_data_object_url
+)
+
+def test_get_collection_name_from_workflow_id():
+    # Test for a known workflow ID
+    workflow_id = "nmdc:wfrqc-11-zbyqeq59.1"
+    assert get_collection_name_from_workflow_id(workflow_id) == "read_qc_analysis_activity_set"
+
+    # Test for an unknown workflow ID
+    workflow_id = "nmdc:unknown-11-zbyqeq59.1"
+    assert get_collection_name_from_workflow_id(workflow_id) == "unknown"
+
+    # Test for a known workflow ID
+    workflow_id = "nmdc:wfmag-11-zbyqeq59.1"
+    assert get_collection_name_from_workflow_id(workflow_id) == "mags_activity_set"
+
+    # Test for a known workflow ID
+    workflow_id = "nmdc:wfmb-11-zbyqeq59.1"
+    assert get_collection_name_from_workflow_id(workflow_id) == "metabolomics_analysis_activity_set"
+
+    # Test for a known workflow ID
```

Comment:
metabolomics can be removed, that collection passes validation with the materalized pattern version of the schema and those foes are having a different naming convention.